crates/matrix-sdk-ui/src/room_list_service/room.rs

```diff
@@ -150,27 +150,32 @@ impl Room {
     }
 
     /// Create a new [`TimelineBuilder`] with the default configuration.
+    ///
+    /// If the room was synced before some initial events will be added to the
+    /// [`TimelineBuilder`].
```

Comment:
Can you add some tests checking for this new behavior?

Reply:
I think this is already tested [here](https://github.com/matrix-org/matrix-rust-sdk/blob/5e662e855db925b380517bd94ef2b3622804a421/crates/matrix-sdk-ui/tests/integration/room_list_service.rs#L2382), it checks the previously synced item is already there: I can create another test if you want to, but it seems kind of redundant. 

Do you want me to test the opposite, which happens when no `SlidingSyncRoom` is found?

Reply:
> Do you want me to test the opposite, which happens when no SlidingSyncRoom is found?

Yes, indeed, since that's the new behavior we're introducing in this PR.

Reply:
Ok, the doc comment mentioned the opposite, so I was a bit confused 😅 . I'll add one, thanks!

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/3979/commits/b472a87ba09723b1b2a65d7b8a66a867f685b686 added a test for this.

Reply:
Thanks!